Latest master does not compile on Windows

<pre>
01:04:02 C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.12.25827\include\xmemory0(945): note: see reference to function template instantiation 'std::pair<int,std::string>::pair<_Ty,std::basic_string<char,std::char_traits<char>,std::allocator<char>>&,0>(_Other1 &&,_Other2) noexcept(false)' being compiled
01:04:02         with
01:04:02         [
01:04:02             _Ty=unsigned __int64,
01:04:02             _Other1=unsigned __int64,
01:04:02             _Other2=std::basic_string<char,std::char_traits<char>,std::allocator<char>> &
01:04:02         ]
01:04:02 C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.12.25827\include\xmemory0(945): note: see reference to function template instantiation 'std::pair<int,std::string>::pair<_Ty,std::basic_string<char,std::char_traits<char>,std::allocator<char>>&,0>(_Other1 &&,_Other2) noexcept(false)' being compiled
01:04:02         with
01:04:02         [
01:04:02             _Ty=unsigned __int64,
01:04:02             _Other1=unsigned __int64,
01:04:02             _Other2=std::basic_string<char,std::char_traits<char>,std::allocator<char>> &
01:04:02         ]
01:04:02 C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.12.25827\include\vector(957): note: see reference to function template instantiation 'void std::_Default_allocator_traits<_Alloc>::construct<_Ty,unsigned __int64,std::basic_string<char,std::char_traits<char>,std::allocator<char>>&>(_Alloc &,_Objty *const ,unsigned __int64 &&,std::basic_string<char,std::char_traits<char>,std::allocator<char>> &)' being compiled
01:04:02         with
01:04:02         [
01:04:02             _Alloc=std::allocator<std::pair<int,std::string>>,
01:04:02             _Ty=std::pair<int,std::string>,
01:04:02             _Objty=std::pair<int,std::string>
01:04:02         ]
01:04:02 C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.12.25827\include\vector(957): note: see reference to function template instantiation 'void std::_Default_allocator_traits<_Alloc>::construct<_Ty,unsigned __int64,std::basic_string<char,std::char_traits<char>,std::allocator<char>>&>(_Alloc &,_Objty *const ,unsigned __int64 &&,std::basic_string<char,std::char_traits<char>,std::allocator<char>> &)' being compiled
01:04:02         with
01:04:02         [
01:04:02             _Alloc=std::allocator<std::pair<int,std::string>>,
01:04:02             _Ty=std::pair<int,std::string>,
01:04:02             _Objty=std::pair<int,std::string>
01:04:02         ]
01:04:02 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\Timers.cpp(607): note: see reference to function template instantiation 'void std::vector<std::pair<int,std::string>,std::allocator<_Ty>>::emplace_back<unsigned __int64,std::basic_string<char,std::char_traits<char>,std::allocator<char>>&>(unsigned __int64 &&,std::basic_string<char,std::char_traits<char>,std::allocator<char>> &)' being compiled
01:04:02         with
01:04:02         [
01:04:02             _Ty=std::pair<int,std::string>
01:04:02         ]
01:04:03 [ 63%] Building CXX object CMakeFiles/pvr.vuplus.dir/src/extract/EpgEntryExtractor.cpp.obj
01:04:03 EpgEntryExtractor.cpp
01:04:03 d:\jenkins\workspace\win-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.h(24): error C2039: 'unique_ptr': is not a member of 'std'
01:04:03 C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.12.25827\include\map(15): note: see declaration of 'std'
01:04:03 d:\jenkins\workspace\win-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.h(24): error C2065: 'unique_ptr': undeclared identifier
01:04:03 d:\jenkins\workspace\win-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.h(24): error C2275: 'VUPLUS::IExtractor': illegal use of this type as an expression
01:04:03 d:\jenkins\workspace\win-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.h(24): error C2974: 'std::vector': invalid template argument for '_Ty', type expected
01:04:03 C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.12.25827\include\vector(658): note: see declaration of 'std::vector'
01:04:03 d:\jenkins\workspace\win-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.h(24): error C3203: 'allocator': unspecialized class template can't be used as a template argument for template parameter '_Alloc', expected a real type
01:04:03 d:\jenkins\workspace\win-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.h(24): error C2059: syntax error: '>'
01:04:03 d:\jenkins\workspace\win-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.h(24): error C2238: unexpected token(s) preceding ';'
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(9): error C2065: 'extractors': undeclared identifier
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(9): error C2228: left of '.emplace_back' must have class/struct/union
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(9): note: type is 'unknown-type'
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(10): error C2065: 'extractors': undeclared identifier
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(10): error C2228: left of '.emplace_back' must have class/struct/union
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(10): note: type is 'unknown-type'
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(19): error C2065: 'extractors': undeclared identifier
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(20): error C3312: no callable 'begin' function found for type 'unknown-type'
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(20): error C3312: no callable 'end' function found for type 'unknown-type'
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(21): error C2065: 'extractor': undeclared identifier
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(21): error C2227: left of '->ExtractFromEntry' must point to class/struct/union/generic type
01:04:03 D:\jenkins\workspace\WIN-64\cmake\addons\build\pvr.vuplus\src\extract\EpgEntryExtractor.cpp(21): note: type is 'unknown-type'
01:04:03 NMAKE : fatal error U1077: 'C:\PROGRA~2\MIB055~1\2017\COMMUN~1\VC\Tools\MSVC\1412~1.258\bin\Hostx64\x64\cl.exe' : return code '0x2'
01:04:03 Stop.
</pre>

I guess this is due to `<memory>`and `<vector>` not included in `EpgEntryExtractor.h`.

You should check all recently added class declarations for correct includes, btw.

Example: `ShowInfoExtractor.h`is missing `<string>`

--- a/src/RecordingReader.h
+++ b/src/RecordingReader.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <ctime>
+#include <string>
 
 #include "libXBMC_addon.h"
 

--- a/src/StreamReader.h
+++ b/src/StreamReader.h
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <string>
+
 #include "IStreamReader.h"
 
 class StreamReader

--- a/src/extract/EpgEntryExtractor.cpp
+++ b/src/extract/EpgEntryExtractor.cpp
@@ -1,13 +1,16 @@
 #include "EpgEntryExtractor.h"
 
+#include "GenreExtractor.h"
+#include "ShowInfoExtractor.h"
+
 using namespace VUPLUS;
 using namespace ADDON;
 
 EpgEntryExtractor::EpgEntryExtractor(const VUPLUS::Settings &settings)
   : IExtractor(settings)
 {
-   extractors.emplace_back(new GenreExtractor(settings));
-   extractors.emplace_back(new ShowInfoExtractor(settings));
+   m_extractors.emplace_back(new GenreExtractor(settings));
+   m_extractors.emplace_back(new ShowInfoExtractor(settings));
 }
 
 EpgEntryExtractor::~EpgEntryExtractor(void)
@@ -16,7 +19,7 @@ EpgEntryExtractor::~EpgEntryExtractor(void)
 
 void EpgEntryExtractor::ExtractFromEntry(VuBase &entry)
 {
-  for (auto& extractor : extractors)
+  for (auto& extractor : m_extractors)
   {
     extractor->ExtractFromEntry(entry);
   }  

--- a/src/extract/EpgEntryExtractor.h
+++ b/src/extract/EpgEntryExtractor.h
@@ -1,9 +1,10 @@
 #pragma once
 
+#include <memory>
 #include <string>
+#include <vector>
 
-#include "GenreExtractor.h"
-#include "ShowInfoExtractor.h"
+#include "IExtractor.h"
 #include "../client.h"
 
 #include "libXBMC_addon.h"
@@ -21,7 +22,7 @@ public:
   void ExtractFromEntry(VuBase &entry);
 
 private:
-  std::vector<std::unique_ptr<IExtractor>> extractors;  
+  std::vector<std::unique_ptr<IExtractor>> m_extractors;  
 };
 
 } //namespace VUPLUS

--- a/src/extract/ShowInfoExtractor.h
+++ b/src/extract/ShowInfoExtractor.h
@@ -3,6 +3,8 @@
 #include "IExtractor.h"
 #include "../client.h"
 
+#include <regex>
+#include <string>
 #include <vector>
 
 #include "libXBMC_addon.h"

